src/main/java/com/icthh/xm/uaa/security/DomainTokenServices.java

```diff
@@ -189,8 +190,16 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
             throw new InvalidGrantException("Invalid refresh token: " + refreshTokenValue);
         }
 
-        OAuth2RefreshToken refreshToken = tokenStore.readRefreshToken(refreshTokenValue);
+        OAuth2RefreshToken refreshToken;
+        try {
+            refreshToken = tokenStore.readRefreshToken(refreshTokenValue);
+        } catch (InvalidTokenException e) {
+            log.warn("Could not refresh token, invalid refresh token: {}", refreshTokenValue);
```

Comment:
why don't  you print the original invalid token e.getMessage to log? i think this information is usseful.